src/s3.js

```diff
@@ -25,7 +25,11 @@ module.exports = function s3UploadFactory(knoxOpt, opt) {
 
 	opt.headers = opt.headers || {};
 
-	const client = knox.createClient(knoxOpt);
+	const client = new AWS.S3({
+		accessKeyId: knoxOpt.key,
```

Comment:
nit: called knox and not creds

Reply:
Haha yeah I know... I was trying to keep things mostly "the same" for ease of review. A major refactor of how all these options get passed around is definitely due.